contracts/test/fixture/_pool_tilt.js

```diff
@@ -0,0 +1,142 @@
+const hre = require("hardhat");
+const { ethers, waffle } = hre;
+const { BigNumber } = ethers;
+const { impersonateAndFundContract, mintWETH } = require("./_fixture");
+const addresses = require("../../utils/addresses");
+
+const { ousdUnits } = require("../helpers");
+
+const { factoryCreatePool } = require("./_pools");
+
+const attackerAddress = "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045";
+let sAttacker;
+let pool;
+
+async function getPoolConfiguration(fixture, pool) {
+  const { rEthBPT, sfrxETHwstETHrEthBPT } = fixture;
+
+  if (!!rEthBPT && pool.address == rEthBPT.address) {
+    return {
+      config: {
+        platform: "balancer",
+        poolType: "metaStable",
+        poolId: await fixture.balancerREthStrategy.balancerPoolId(),
+        pool: rEthBPT,
+        // important for Balancer type pools order of these must match the Balancer pool
+        assetAddressArray: [fixture.reth.address, fixture.weth.address],
+        bptToken: rEthBPT,
+        balancerVault: fixture.balancerVault,
+      },
+    };
+  } else if (
+    !!sfrxETHwstETHrEthBPT &&
+    pool.address == sfrxETHwstETHrEthBPT.address
+  ) {
+    return {
+      config: {
+        platform: "balancer",
+        poolType: "composableStable",
+        poolId: await fixture.balancerSfrxWstRETHStrategy.balancerPoolId(),
+        pool: sfrxETHwstETHrEthBPT,
+        // important for Balancer type pools order of these must match the Balancer pool
+        assetAddressArray: [
+          sfrxETHwstETHrEthBPT.address,
+          addresses.mainnet.wstETH,
+          addresses.mainnet.sfrxETH,
+          addresses.mainnet.rETH,
+        ],
+        bptToken: sfrxETHwstETHrEthBPT,
+        balancerVault: fixture.balancerVault,
+      },
+    };
+  }
+}
+
+async function _fundAttackerOption({
+  asset, // attacking asset
+  assetAmount, // required funded asset amount to perform tilt
+  fixture,
+}) {
+  const attackerEthBalance = await waffle.provider.getBalance(attackerAddress);
```

Comment:
`hre.network.provider` or `hre.ethers.provider` should work. `waffle` is an unused dependency we should get rid of 😅

Reply:
have changed the way provider is accessed, since it is cleaner. The waffle is actually used in test infrastructure. Removing it failed a test for me: 
![Screenshot 2023-10-19 at 18 17 30](https://github.com/OriginProtocol/origin-dollar/assets/579910/1b5fe330-512f-4152-80f7-98fd583e9248)

It also seems to contain chai matchers and fixtures: https://www.npmjs.com/package/ethereum-waffle

Reply:
Thanks, I'll deal with the waffle in a different PR